bids-validator/src/schema/applyRules.ts

```diff
@@ -446,92 +446,90 @@ function evalJsonCheck(
   // Count on other JSON rules to use selectors to match the correct files
   if (context.extension === '.json' && sidecarRule) return
 
-  const json = sidecarRule ? context.sidecar : context.json
+  const json: Record<string, any> = sidecarRule ? context.sidecar : context.json
   for (const [key, requirement] of Object.entries(rule.fields)) {
-    const severity = getFieldSeverity(requirement, context)
     // @ts-expect-error
     const metadataDef = schema.objects.metadata[key]
     const keyName: string = metadataDef.name
-    if (severity && severity !== 'ignore' && !(keyName in json)) {
-      if (requirement.issue?.code && requirement.issue?.message) {
-        context.dataset.issues.add({
-          code: requirement.issue.code,
-          subCode: keyName,
-          location: context.path,
-          severity,
-          rule: schemaPath,
-        }, requirement.issue.message)
-      } else {
-        let code
-        if (severity === 'error') {
-          code = sidecarRule ? 'SIDECAR_KEY_REQUIRED' : 'JSON_KEY_REQUIRED'
-        } else if (severity === 'warning') {
-          code = sidecarRule ? 'SIDECAR_KEY_RECOMMENDED' : 'JSON_KEY_RECOMMENDED'
-        }
-        if (code) {
+    const value = json[keyName]
+    if (value === undefined) {
+      const severity = getFieldSeverity(requirement, context)
+      if (severity && severity !== 'ignore') {
+        if (requirement.issue?.code && requirement.issue?.message) {
           context.dataset.issues.add({
-            code,
+            code: requirement.issue.code,
             subCode: keyName,
             location: context.path,
             severity,
-            issueMessage: `missing ${keyName}`,
             rule: schemaPath,
-          })
+          }, requirement.issue.message)
+        } else {
+          let code
+          if (severity === 'error') {
+            code = sidecarRule ? 'SIDECAR_KEY_REQUIRED' : 'JSON_KEY_REQUIRED'
+          } else if (severity === 'warning') {
+            code = sidecarRule ? 'SIDECAR_KEY_RECOMMENDED' : 'JSON_KEY_RECOMMENDED'
+          }
+          if (code) {
+            context.dataset.issues.add({
+              code,
+              subCode: keyName,
+              location: context.path,
+              severity,
+              rule: schemaPath,
+            })
+          }
```

Comment:
Just to shave off a few lines:

```suggestion
          const keyType = sidecarRule ? 'SIDECAR_KEY' : 'JSON_KEY'
          const level = severity === 'error' ? 'REQUIRED' : 'RECOMMENDED'
          context.dataset.issues.add({
            `${keyType}_${level}`,
            subCode: keyName,
            location: context.path,
            severity,
            rule: schemaPath,
          })
```